homeassistant/components/tfl/sensor.py

```diff
@@ -0,0 +1,162 @@
+
+from . import TfLConfigEntry
+from .common import call_tfl_api
+from .const import CONF_STOP_POINTS, DOMAIN
+
+ATTR_NEXT_ARRIVALS = "all"
+ATTR_NEXT_THREE_ARRIVALS = "next_3"
+
+RAW_ARRIVAL_LINE_NAME = "lineName"
+RAW_ARRIVAL_DESTINATION_NAME = "destinationName"
+RAW_ARRIVAL_TIME_TO_STATION = "timeToStation"
+
+SCAN_INTERVAL = timedelta(seconds=30)
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: TfLConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the TfL sensor."""
+    stop_point_api = entry.runtime_data.stop_point_api
+
+    conf = entry.options
+
+    stop_point_ids: list[str] = conf[CONF_STOP_POINTS]
+
+    try:
+        stop_point_infos = await call_tfl_api(
+            hass, stop_point_api.getByIDs, stop_point_ids, False
+        )
+        devices = []
+        if isinstance(stop_point_infos, list):
+            for idx, stop_point_id in enumerate(stop_point_ids):
+                devices.append(
+                    StopPointSensor(
+                        stop_point_api,
+                        stop_point_infos[idx]["commonName"],
+                        stop_point_id,
+                        entry.entry_id,
+                    )
+                )
+        else:
+            devices.append(
+                StopPointSensor(
+                    stop_point_api,
+                    stop_point_infos["commonName"],
+                    stop_point_ids[0],
+                    entry.entry_id,
+                )
+            )
+
+        async_add_entities(devices, True)
+
+    except HTTPError as exception:
+        error_code = exception.code
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with HTTP error_code=%s, entities will not be created",
+            stop_point_ids,
+            error_code,
+        )
+    except URLError as exception:
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with URLError reason=%s, entities will not be created",
+            stop_point_ids,
+            exception.reason,
+        )
+
+
+class StopPointSensor(SensorEntity):
+    """Representation of a TfL StopPoint as a Sensor."""
+
+    _attr_attribution = "Powered by TfL Open Data"
+    _attr_icon = "mdi:bus"
+
+    def __init__(
+        self, stop_point_api: stopPoint, name: str, stop_point_id: str, entry_id: str
+    ) -> None:
+        """Initialize the TfL StopPoint sensor."""
+        # super().__init__(coordinator)
+        self._name = name
+        self._attr_name = name
+        self._attr_unique_id = stop_point_id
+        self._attr_device_info = DeviceInfo(
+            name="TfL",
+            identifiers={(DOMAIN, entry_id)},
+            entry_type=DeviceEntryType.SERVICE,
+        )
+
+        self._stop_point_api = stop_point_api
+        self._stop_point_id = stop_point_id
+
+    async def async_update(self) -> None:
```

Comment:
Out of curiosity, why does the sensor focus on arrivals, isn't departure time typically more useful?

Reply:
Interesting question! The API ([example for Oxford Circus](https://api.tfl.gov.uk/StopPoint/940GZZLUOXC/arrivals)) is only providing the `time_to_station` and the API is `/arrivals`.
Which kinda makes sense… in a metro area like London, the tube and the bus will go as soon as it's ready. So the departure time is a function of how many people are getting on and off the tube/bus and can't be predicted. The arrival time is more predictable, and as a traveller you want to be there for the arrival time.